src/MongoDB.Driver.Core/Core/Clusters/Cluster.cs

```diff
@@ -308,7 +308,9 @@ protected void UpdateClusterDescription(ClusterDescription newClusterDescription
             }
 
             OnDescriptionChanged(oldClusterDescription, newClusterDescription);
-            oldDescriptionChangedTaskCompletionSource.TrySetResult(true);
+
+            // TODO: use RunContinuationsAsynchronously instead once we require a new enough .NET Framework
+            Task.Run(() => oldDescriptionChangedTaskCompletionSource.TrySetResult(true));
```

Comment:
Using Task.Run schedules the call to TrySetResult to execute later on a separate thread.

Reply:
Not sure, how better do it, but as I understand this TODO comment will be lost as soon as this ticket will be closed until somehow we will work with this piece of code again. Maybe we can create a ticket for this TODO? Which will have a description like pending of bumping up the supported .net version?

Reply:
I agree. See this ticket which was created yesterday:

https://jira.mongodb.org/browse/CSHARP-2573